programs/lfx-mentorship/2023/02-Jun-Aug/project_ideas.md

```diff
@@ -57,4 +57,22 @@ Test coverage could increase
 - Mentor(s): Yuri Shkuro (@yurishkuro, github@ysh.us)
 - Upstream Issue: https://github.com/jaegertracing/jaeger/issues/3381
 
+### Knative
```

Comment:
Can we add this idea too (@evankanderson ? I can be a co-mentor or maybe @creydr ?
```
#### Eventing Sender Identity

- Description: Leveraging Kubernetes Service Account identity and OAuth audiences, users should be able to configure Knative Eventing components to authenticate CloudEvent deliveries using the identity of the Subscription, Trigger, or Source. Additionally, Brokers and Channels can leverage the OAuth identity associated with incoming CloudEvents to implement policy.
- Expected outcome: At least the following components are able to use service accounts for authentication: in-memory-channel, multitenant broker, apiserver source, ping source. Ideally, the primitives from this project could be re-used by other channel and broker implementations.
- Recommended Skills: Golang, Kubernetes, OAuth or OIDC
- Mentor(s): Evan Anderson @evankanderson
- Expected project size: 350h
- Difficulty: Medium
- Upstream Issue (URL): https://github.com/knative/eventing/issues/5047
```

Reply:
I also pinged @evankanderson on Slack. Good call @pierDipi

Reply:
I would be happy to co-mentor on this topic